### PR TITLE
[stable/kube-state-metrics] Add service type

### PR DESCRIPTION
**Pull request by thedebugger**

### CODE REVIEW

**Comment by thedebugger** [comment]
Hi, can i get some eyes on this PR? Thanks!

File: stable/kube-state-metrics/templates/service.yaml [hunk 1]
```diff
@@ -12,11 +12,18 @@ metadata:
     prometheus.io/scrape: '{{ .Values.prometheusScrape }}'
   {{- end }}
 spec:
+  type: {{ .Values.service.type }}
```
Comment (mgoodness) [inline stable/kube-state-metrics/templates/service.yaml]: Add double-quotes around templates that simply echo user-provided strings.

File: stable/kube-state-metrics/templates/service.yaml [hunk 2]
```diff
@@ -12,11 +12,18 @@ metadata:
...
   ports:
   - name: "http"
     protocol: TCP
     port: {{ .Values.service.port }}
+    {{- if .Values.service.nodePort }}
```
Comment (mgoodness) [inline stable/kube-state-metrics/templates/service.yaml]: Nit: `if` and `end` statements should be indented one level less than the lines they wrap.

File: stable/kube-state-metrics/templates/service.yaml [hunk 3]
```diff
@@ -12,11 +12,18 @@ metadata:
...
+    nodePort: {{ .Values.service.nodePort }}
+    {{- end }}
     targetPort: 8080
+  {{- if .Values.service.loadBalancerIP }}
+  loadBalancerIP: {{ .Values.service.loadBalancerIP }}
```
Comment (mgoodness) [inline stable/kube-state-metrics/templates/service.yaml]: Add double-quotes around templates that simply echo user-provided strings.

File: stable/kube-state-metrics/templates/service.yaml [hunk 4]
```diff
@@ -12,11 +12,18 @@ metadata:
...
+  type: {{ .Values.service.type }}
   ports:
   - name: "http"
     protocol: TCP
     port: {{ .Values.service.port }}
+    {{- if .Values.service.nodePort }}
+    nodePort: {{ .Values.service.nodePort }}
+    {{- end }}
     targetPort: 8080
+  {{- if .Values.service.loadBalancerIP }}
```
Comment (mgoodness) [inline stable/kube-state-metrics/templates/service.yaml]: Nit: as above.

File: stable/kube-state-metrics/values.yaml [hunk 5]
```diff
@@ -6,3 +6,7 @@ image:
   pullPolicy: IfNotPresent
 service:
   port: 8080
+  # Default to clusterIP for backward compatibility
+  type: ClusterIP
+##nodePort: 30001
```
Comment (mgoodness) [inline stable/kube-state-metrics/values.yaml]: Un-comment and set to `0` (zero)

File: stable/kube-state-metrics/values.yaml [hunk 6]
```diff
@@ -6,3 +6,7 @@ image:
...
+##loadBalancerIP: 78.11.24.19
```
Comment (mgoodness) [inline stable/kube-state-metrics/values.yaml]: Un-comment and set to `""`

**Review by mgoodness — CHANGES_REQUESTED** [review]
Mostly nits, but also needs rebase

**Comment by thedebugger** [comment]
Rebased and incorporated code review feedback

**Comment by thedebugger** [comment]
@mgoodness can you look at this today if you have time?

**Comment by thedebugger** [comment]
hello @kubernetes/charts-maintainers, it is been a while that someone has looked at this PR. I'm reaching a point where i have to fork and manage my repo but I don't want to do that.

**Comment by thedebugger** [comment]
Did another rebase

**Pull request merged**